src/main/kotlin/controllers/HeadRequestController.kt

```diff
@@ -0,0 +1,37 @@
+package controllers
+
+import httpstatus.HttpStatus
+import response.ResponseBuilder
+
+class HeadRequestController : Controller {
+    private var statusCode = HttpStatus.OK
+    private val headers: Map<String, String> = mapOf("Allow" to "HEAD, OPTIONS")
+    private lateinit var requestBody: String
+    private lateinit var method: String
+
+    override fun action(): ResponseBuilder {
+        if (checkForInvalidMethod()) {
+            statusCode = HttpStatus.NotAllowed
+            return ResponseBuilder(statusCode, headers = headers)
+        }
+        return ResponseBuilder(statusCode)
+    }
+
+    override fun setBody(body: String) {
+        requestBody = body
+    }
+
+    override fun setHttpMethod(httpMethod: String) {
+        method = httpMethod
+    }
+
+    private fun checkForInvalidMethod(): Boolean {
+        val allowedMethods = headers["Allow"]
+        if (allowedMethods != null) {
+            if (!allowedMethods.contains(method)) {
+                return true
+            }
+        }
+        return false
+    }
```

Comment:
Is this behavior specific to this controller? It seems like it could be used by all controllers.

Reply:
I created a `MethodNotAllowedController` to replace the need for each controller having to check for invalid methods